### PR TITLE
Paper: Spatio-temporal analysis of socioeconomic neighborhoods

### PR DESCRIPTION
**Pull request by sjsrey**
If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.

### CODE REVIEW

**Comment by msarahan** [comment]
I'm far from an expert here, but I found this paper interesting to read as someone interested in demographic trends for studying the effects of gerrymandering.  I have nothing to recommend or add to this paper, other than to say nice work.

File: papers/serge_rey/serge_rey.rst [hunk 1]
```diff
@@ -0,0 +1,799 @@
...
+:author: Wei Kang 
+:email: 
+:institution: Center for Geospatial Sciences, University of California, Riverside 
+
+:bibliography: reference
+
+
+
+-------------------------------------------------------
+Spatio-temporal analysis of socioeconomic neighborhoods
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: Title is too broad. Since this is a paper on the OSLNAP tool, and not the analysis of socioeconomic neighborhoods per se, why not include that in the title?

File: papers/serge_rey/serge_rey.rst [hunk 2]
```diff
@@ -0,0 +1,799 @@
...
+
+The neighborhood effects literature represents a wide span of the social
+sciences broadly concerned with the influence of spatial context on social
+processes. From the study of segregation dynamics, the relationships between the
+built environment and health outcomes, to the impact of concentrated poverty on
+social efficacy, neighborhoods are a central construct in empirical work. From a
+dynamic lens, neighborhoods experience changes not only in their socioeconomic
+composition, but also in spatial extent; however, the literature has ignored the
+latter source of change. In this paper, we discuss the development of novel,
+spatially explicit approaches to the study of longitudinal neighborhood dynamics
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: "we discuss the development of novel, spatially explicit approaches..." sounds too broad to me. "We discuss the development of a novel, spatially explicit tool, OSLNAP..." sounds more accurate and appealing to the reader.

File: papers/serge_rey/serge_rey.rst [hunk 3]
```diff
@@ -0,0 +1,799 @@
...
+different interpolation methods for housing prices than for the share of
+unemployed residents, than for total population; not only because the
+researcher may wish to treat rates and counts separately, but also
+because different auxiliary information might be applicable for
+different types of variables.
+
+In a prototypical workflow, ``harmonize`` permits the end-user to:
+
+-  compile and query a spatiotemporal database using either local data or
+   connections to  pubilic data services
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: Typo: pubilic

File: papers/serge_rey/serge_rey.rst [hunk 4]
```diff
@@ -0,0 +1,799 @@
...
+implications of including space as a formal component in their
+clustering models.
+
+In OSLNAP, the ``cluster`` module leverages the scientific python ecosystem,
+building from scikit-learn :cite:`scikit-learn`, geopandas :cite:`geopandas`,
+and PySAL :cite:`Rey14`. Using input from the Data Layer, the ``cluster`` module
+allows researchers to develop neighborhood typologies based on either attribute
+similarity (the geodemographic approach) or attribute similarity with
+incorporated spatial dependence (the regionalization approach). Given a
+space-time data set, the ``cluster`` module allows users to cluster (a) a single
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: What "the cluster module allows..." is a little unclear. Can you extend this a little bit? For example, what do you mean by clustering a single time period? I believe clustering neighborhoods at a given time? These capabilities deserve a little more detailed explanation with examples to illustrate the idea better.

File: papers/serge_rey/serge_rey.rst [hunk 5]
```diff
@@ -0,0 +1,799 @@
...
+socioeconomic composition, while we and others have argued that changes in
+geographic footprints are also substantively interesting :cite:`rey2011`.
+Therefore, this component draws upon recent methodological developments from
+spatial inequality dynamics and implements two broad sets of spatially explicit
+analytics to provide deeper insights into the evolution of socioeconomic
+processes and the interaction between these processes and geographic structure.
+
+Both sets of analytics operate on time series of neighborhood types; they each take
+as input a set of spatial units of analysis (e.g. census tracts) that have
+been assigned a categorical variable for each point in time (e.g.the output of
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: Space after e.g.

File: papers/serge_rey/serge_rey.rst [hunk 6]
```diff
@@ -0,0 +1,799 @@
...
+one sequence to another using a combination of operations including
+substitution, insertion, deletion and transposition.
+The similarity matrix is then used as the input for another round of clustering
+to derive a typology of neighborhood trajectory to produce several sequences
+of neighborhood types typically
+happening in a particular order :cite:`delmelle2016`.
+It should be noted here that the operation costs are not necessarily
+identical. Rather, they can be manipulated so that the resulted similarity
+measure reflects the exact characteristics of the neighborhood sequence we are
+interested in. For example, the cost of substution may capture units' socioeconomic
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: Typo: substution

File: papers/serge_rey/serge_rey.rst [hunk 7]
```diff
@@ -0,0 +1,799 @@
...
+dissimilarity in contemporary experience :cite:`li2018` while operations
+including insertion, deletion and
+transposition can be assigned such an expensive cost that it is highly unlikely
+they will be allowed in the OM process .
+We allows for various cost functions and further extend the definition
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: Typo: We allows

File: papers/serge_rey/serge_rey.rst [hunk 8]
```diff
@@ -0,0 +1,799 @@
...
+   :figclass: w
+
+   Neighborhood Types in LA using SKATER. :label:`f:skater`
+
+Despite the fact that clusters are independent from one year to the next
+(and thus, we lack appropriate space in this text for describing the
+SKATER results for each year) comparing the results over time
+nonetheless yield some interesting insights. Regardless of the changing
+spatial and demographic structure of the Los Angeles region, some of the
+of the neighborhood boundaries identified are remarkably stable, such as
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: Typo: some of the of the

File: papers/serge_rey/serge_rey.rst [hunk 9]
```diff
@@ -0,0 +1,799 @@
...
+and their social re-definition as developed by different residential
+groups :cite:`wolf2018`. Irrespective of its particular use, the
+regionalization approach presents neighborhood researchers with another
+critical tool for understanding the bi-directional relationship between
+people and places.
+
+In each of the sample analyses presented above, we use ``OSLNAP`` to
+derive a set of neighborhood clusters or types that can be used to
+analyze the demographic makeup of places over time. In some cases, these
+maps can serve as foundations for descriptive analyses or analyzed as
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: "or be analyzed"

File: papers/serge_rey/serge_rey.rst [hunk 10]
```diff
@@ -0,0 +1,799 @@
...
+
+The ``change``
+module can provide insights into the nature of neighborhood change in
+the Los Angeles metropolitan area. We utilize the neighborhood types
+for all census tracts of the Los Angeles metropolitan area across four
+census years identified by selected clustering algorithms in
+the former section as the input for the ``change`` module. Among the
+three clustering algorithms, SKATER was applied to each cross
+section of census tracts independently yielding clusters which are not
+directly comparable over time. Thus, we focuses only on the
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: Typo: we focuses

File: papers/serge_rey/serge_rey.rst [hunk 11]
```diff
@@ -0,0 +1,799 @@
...
+series of neighborhood types and counts how many transitions between
+any pair of neighborhood types across immediate consecutive census years :math:`(t,t+10)`
+(or :math:`(t,t+5)` for 2010-2015)
+which are further organized into a :math:`(k,k)` transition count matrix
+:math:`\pmb{N}`. Adopting
+the maximum likelihood estimator for the first-order Markov transition
+probability as shown in Equation (:ref:`mle`), a :math:`(k,k)` transition
+probability matrix can thus be constructed providing the insights in the
+underlying dynamics of neighborhood change. The :math:`(6,6)` and the
+:math:`(14,14)` transition probability matrices for ward and affinity
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: Typo: ward -> Ward

File: papers/serge_rey/serge_rey.rst [hunk 12]
```diff
@@ -0,0 +1,799 @@
...
+propagation clusters are estimated and visualized in Fig. :ref:`f:wardapm` where
+the color in grid :math:`(i,j)` represents the probability of transitioning from
+neighborhood type :math:`i` to :math:`j` in the next census year. It is obvious
+that both transition probability matrices are characterized by large diagonal
+entries, indicating a certain level of neighborhood stability for the focal four
+census years. This is especially true for the ward neighborhood type 4 which is
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: Typo: ward -> Ward

File: papers/serge_rey/serge_rey.rst [hunk 13]
```diff
@@ -0,0 +1,799 @@
...
+spatial lag. Therefore,
+:math:`k` exhaustive and mutually exclusive subsamples are constructed based
+on the spatial lag at :math:`t`, from which :math:`k` :math:`(k,k)` transition probability
+matrices are estimated based on Equation (:ref:`mle`). Fig. :ref:`f:smward`
+displays the spatial Markov transition probability matrices for Ward neighborhood types.
+It should be noted that the interpretation with these conditional transition probabilities
+should proceed with caution as the increased number of parameters to be estimated
+here could lead to large standard errors for some estimates. For example, the
+:math:`(0,0)` entry in the subplot of Spatial Lag 3 is :math:`1`. The tendency
+of interpreting the 100 percent to be tracts "perfectly stuck at" ward neighborhood type 0 if the spatial lag
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: Typo: ward -> Ward

File: papers/serge_rey/serge_rey.rst [hunk 14]
```diff
@@ -0,0 +1,799 @@
...
+of social/spatial volatility as a move from "minority working class" to "white upper class,"
+as might happen during rapid gentrification.
+
+As such, we can also weight the edit distance based on how "expensive" the edit
+is in terms of the clustering distance. Using this weighting method, not all
+transitions from white working class to white upper class will be treated the
+same: observations that are "almost" white upper class but not quite will be
+considered more similar to white upper class tracts. But, since a reassignment
+is still involved, there will still be a cost associated with that edit.
+Clusterings for both the raw hamming edit distance and the weighted hamming edit
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: Typo: hamming -> Hamming

File: papers/serge_rey/serge_rey.rst [hunk 15]
```diff
@@ -0,0 +1,799 @@
...
+distances over sociodemographic sequences are shown in Figure :ref:`f:trajclust`
+using :cite:`wolf2018`. Broadly speaking, the assignments between the two
+clustering methods are strongly related (with an adjusted Rand index of .68),
+but macro-level distinctions between assignment structures are visible,
+particularly in in the areas of central northern LA near the Hollywood Hills, as
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: Typo: in in

File: papers/serge_rey/serge_rey.rst [hunk 16]
```diff
@@ -0,0 +1,799 @@
...
+cases.
+
+In general, since decisions about how to operationalize these distance metrics, which initial
+clusterings to use to generate sequences, and the final clustering algorithms all may have 
+a significant effect on the resulting clusters discovered, it becomes important to support many
+different options, configurations, and analysis workflows. As discussed above, 
+the differences in "final" results can also impact the substantive understandings of how 
+neighborhood dynamics are operating in an area under study. Thus, providing the user with 
+many ways to flexibly reparameterize these complex questions about space-time similarity
+will enhance and improve the usefulness of ``OSLNAP`` for reproducible urban science.
```
Comment (FatihAkici) [inline papers/serge_rey/serge_rey.rst]: "Enhance and improve" sounds redundant.

**Review by FatihAkici — CHANGES_REQUESTED** [review]
Overall very good paper and tool. Thanks for your contribution to SciPy 2018! Other than a few comments that I left in-line, my two main comments regarding the overall quality of the paper are:

1. The title and abstract are too broad. It would be better both for the paper and reader if OSLNAP, the center of this paper, was mentioned in both title and abstract.
2. Figures are unreadable. Please make them bigger.

Thank you!

**Comment by AngelosPsy** [comment]
#  Spatio-temporal analysis of socioeconomical neighborhoods,

In their manuscript named Spatio-temporal analysis of socioeconomical neighborhoods, Rey, Knaap, Han, Wolf, & Kang present a program for analyses of socioecononomic neighborhoods. Overall, the manuscript is strong, with clear examples on what the module can do. I would like to congratulate the authors for delivering such a solid manuscript. My comments refer mostly to textual changes so that it will enable the text to be a bit easier to follow by individuals outside the particular field (such as myself). I list all my suggestions below. 


# Written quality

- "...however, neighborhoods remain understudied because researchers lack appropriate analytical tools for under- standing their evolution through time and space.": I reckon that somebody could also think of other reasons why neighborhoods are understudied than just the lack of appropriate tools. I would rephrase this to something like "...however, neighborhoods remain understudied. One of the reasons for that is because".

- Some of the included lists could be rewritten so that they enable them to be read more easily. To illustrate, on page 2 the authors write: "Compile and query a spatiotemporal database using either local data or connectionsÉÓ and then lists the types of queries. I think it would be better for the text's comprehension, as well as for saving some space so that the some of the figures could be enlarged (see below), if the lists were included in the main text. This also goes to the list on p. 3.

- (p. 4) "We allows". Is this a typo? 

- (p. 7) Apart from the first paragraph, the conclusion section is mostly a list of future directions. It would be better if this section is rephreased to _future directions_ with a separate conclusion section following it.

# Technical Content

- I find the technical content excellent.

# Novelty and impact

- Excellent work by the authors in terms of what is new with the software and why the particular software is important to the community (e.g., end of abstract, p. 2, third paragraph).


# Verifiability

- No code snippets are included. The paper includes plenty of code references and rough descriptions about what the code can do. Maybe the authors want to check again the verifiability requirements that each paper should meet (https://github.com/scipy-conference/scipy_proceedings/blob/master/review_criteria.md#verifiability). I understand that due to the length of the paper it is hard to include source code but maybe the authors should consider that.

# Figures

- Figures 2, 3, 4, 7, and 8 are quite small, making it difficult to see them.

File: papers/serge_rey/serge_rey.rst [hunk 17]
```diff
@@ -0,0 +1,799 @@
...
+Longitudinal Tract Database (LTDB) :cite:`logan_2014`. The developers of
+these products have published studies verifying the accuracy
+of their respective data, but those claims have gone untested because
+external researchers are unable to fully replicate the underlying methodology.
+
+To overcome the issues outlined above, ``OSLNAP`` provides a suite of
+methods for conducting areal interpolation and boundary
+harmonization in the ``harmonize`` module. It leverages ``geopandas``
+and ``PySAL`` for managing data and performing geospatial operations,
+and the pydata stack for attribute calculations :cite:`rey2009`. The
```
Comment (TomAugspurger) [inline papers/serge_rey/serge_rey.rst]: pydata -> PyData?

**Review by TomAugspurger — COMMENTED** [review]
Like Michael, this is outside my area of expertise and I don't have anything to add. It was an enjoyable read.

**Comment by deniederhut** [comment]
Hi @sjsrey ! Do you have some updates on the paper?

**Comment by sjsrey** [comment]
@deniederhut Yes - planning on this weekend for the updates.

File: papers/serge_rey/serge_rey.rst (hunk 10 above)
Comment (sjsrey) [reply papers/serge_rey/serge_rey.rst]: fixed

File: papers/serge_rey/serge_rey.rst (hunk 1 above)
Comment (sjsrey) [reply papers/serge_rey/serge_rey.rst]: Thank you for this suggestion. We agree that the title and abstract needed to be more explicitly linked to the tool, and we have changed both in this regard.

File: papers/serge_rey/serge_rey.rst (hunk 2 above)
Comment (sjsrey) [reply papers/serge_rey/serge_rey.rst]: Incorporated.

File: papers/serge_rey/serge_rey.rst (hunk 3 above)
Comment (sjsrey) [reply papers/serge_rey/serge_rey.rst]: Corrected.

File: papers/serge_rey/serge_rey.rst (hunk 11 above)
Comment (sjsrey) [reply papers/serge_rey/serge_rey.rst]: fixed

File: papers/serge_rey/serge_rey.rst (hunk 12 above)
Comment (sjsrey) [reply papers/serge_rey/serge_rey.rst]: Fixed

File: papers/serge_rey/serge_rey.rst (hunk 13 above)
Comment (sjsrey) [reply papers/serge_rey/serge_rey.rst]: fixed

File: papers/serge_rey/serge_rey.rst (hunk 14 above)
Comment (sjsrey) [reply papers/serge_rey/serge_rey.rst]: Fixed

File: papers/serge_rey/serge_rey.rst (hunk 15 above)
Comment (sjsrey) [reply papers/serge_rey/serge_rey.rst]: Fixed

File: papers/serge_rey/serge_rey.rst (hunk 16 above)
Comment (sjsrey) [reply papers/serge_rey/serge_rey.rst]: Agreed.

**Comment by deniederhut** [comment]
Dear @msarahan , @AngelosPsy , and @AngelosPsy do you feel that this paper is now ready for inclusion in the proceedings?

**Comment by AngelosPsy** [comment]
Dear @deniederhut. Thank you for your message. Will be abroad tomorrow. Will check the revised paper on Wednesday. Is that OK?

**Comment by deniederhut** [comment]
Sure thing. And sorry, one of those @s was supposed to be for @FatihAkici

**Comment by FatihAkici** [comment]
@sjsrey Thanks for the revision! The paper is looking much nicer and better-organized. However the "we focuses" typo is still there, and some images are still hard to see (Figures 2, 3, 4, 8). The dates on the Fig. 4 are tiny, and the headers&legends on Fig. 8 are impossible to see. And also, only the figures 2, 3, 5 and 6 are mentioned in the text. Do you mind mentioning the rest and their contexts? Finally, the "1" in the title looks incorrect, can you include that funding acknowledgement as a footnote without that "1" or open an Acknowledgement section at the end?

@deniederhut, once the points I just noted are addressed, and @AngelosPsy gives the thumbs-up, we should be good to go!

**Comment by AngelosPsy** [comment]
Dear @sjsrey, thank you for all the revisions. The paper is much better now. Could you just add a short conclusion paragraph at the end -- it was mentioned in my initial review. If not, please let me know.

**Comment by sjsrey** [comment]
@AngelosPsy Thank you (and all the reviewers) for your careful reading of our work and very helpful comments. We’ve added the short conclusion paragraph and cut a little elsewhere to remain within the 8 page limit.

**Comment by AngelosPsy** [comment]
Dear @sjsrey. Thank you for addressing my final comment. For me the paper is good to go now. @FatihAkici  if your comments are addressed sufficiently, we can notify the committee.

**Comment by FatihAkici** [comment]
Dear @AngelosPsy thanks so much for your review, and thanks again for the paper, dear @sjsrey! @proceedings, the paper is ready to be merged.

**Review by FatihAkici — APPROVED** [review]

**Review by deniederhut — CHANGES_REQUESTED** [review]
Hi @sjsrey ! It looks like these citations are missing their DOIs. Could I ask you to add them in before we merge this paper?

**Comment by sjsrey** [comment]
@deniederhut I added the doi for the one dataset that is used in the illustration. If we add doi's for all the citations were are going to go over the 8 page limit. Should we add doi's to every citation, or is it required that the data sources be referenced in this fashion?

**Comment by deniederhut** [comment]
The legal agreement that lets us issue a DOI for your paper requires that
your paper reference other DOIs where possible.

**Comment by sjsrey** [comment]
@deniederhut I can add them where possible, but this will mean probably 9 pages of text - is that permissible?

We have added dois to all citations that had them available.

**Comment by deniederhut** [comment]
Thanks for adding the DOIs! We would still prefer that papers stay within the page limit - would it be possible to pull out a paragraph somewhere? Or maybe find a more compact way to represent those lists of neighborhood labels?

**Review by deniederhut — APPROVED** [review]

**Comment by deniederhut** [comment]
Thanks for working with us on the paper! This looks good to me
